packages/teleport/src/Discover/discoverContextProvider.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { DiscoverContext } from './discoverContext';
+
+export const DiscoverReactContext = React.createContext<DiscoverContext>(null);
+
+const DiscoverContextProvider: React.FC<{
+  value: DiscoverContext;
+}> = props => {
+  return <DiscoverReactContext.Provider {...props} />;
+};
+
+export default DiscoverContextProvider;
+
+export function useDiscoverContext() {
+  const ctx = React.useContext(DiscoverReactContext);
+  if (!ctx) {
```

Comment:
I _think_ this won't be reached - if `useDiscoverContext` is used without the context provider being in the parent tree, `React.useContext(DiscoverReactContext)` would throw the error for us